lib/IDE/Refactoring.cpp

```diff
@@ -1463,6 +1463,112 @@ bool RefactoringActionExtractRepeatedExpr::performChange() {
                                           EditConsumer).performChange();
 }
 
+struct CollapsibleNestedIfInfo {
+  IfStmt *OuterIf;
+  IfStmt *InnerIf;
```

Comment:
Does this mean that refactoring will support only 2 if statements initially?

Reply:
I'm personally ok with collapsing only 2 if statements at one refactoring operation, since the users can always do more by invoking the tool multiple times.